DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,166 @@
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateTime))
+                {
+                    return null;
+                }
+
+                return new DateTimeOffset(dateTime.Year, 1, 1, 0, 0, 0, default);
+            }
+
+            return null;
+        }
+
+        public DateTimeObject Redact(DateTimeObject dateObject)
+        {
+            EnsureArg.IsNotNull(dateObject, nameof(dateObject));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateObject.DateValue))
+                {
+                    return null;
+                }
+
+                dateObject.DateValue = new DateTimeOffset(dateObject.DateValue.Year, 1, 1, 0, 0, 0, dateObject.HasTimeZone == true ? dateObject.DateValue.Offset : new TimeSpan(0, 0, 0));
+                return dateObject;
+            }
+
+            return null;
+        }
+
+        public AgeObject Redact(AgeObject age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            return RedactAge(age.AgeInYears()) == null ? null : age;
+        }
+
+        private string RedactDate(string dateString)
+        {
+            if (string.IsNullOrEmpty(dateString))
+            {
+                return null;
+            }
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateString, DeIDGlobalSettings.DateFormat);
+
+            return Radact(date)?.ToString(DeIDGlobalSettings.DateFormat);
+        }
+
+        private string RedactDateTime(string dateTimeString)
+        {
+            if (string.IsNullOrEmpty(dateTimeString))
+            {
+                return null;
+            }
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateTimeString, DeIDGlobalSettings.DateTimeFormat);
+
+            return Radact(date)?.ToString(DeIDGlobalSettings.DateTimeFormat);
+        } 
+
+        private uint? RedactAge(uint age)
+        {
+            return (uint?)RedactAge((decimal)age);
+        }
+
+        private decimal? RedactAge(decimal age)
+        {
+            if (_redactSetting.EnablePartialAgesForRedact)
+            {
+                if (age > AgeThreshold)
+                {
+                    return null;
+                }
+
+                return age;
+            }
+
+            return null;
+        }
+
+        private string RedactPostalCode(string postalCode)
+        {
+            EnsureArg.IsNotNull(postalCode, nameof(postalCode));
+
+            if (_redactSetting.EnablePartialZipCodesForRedact)
+            {
+                if (_redactSetting.RestrictedZipCodeTabulationAreas != null && _redactSetting.RestrictedZipCodeTabulationAreas.Any(x => postalCode.StartsWith(x)))
+                {
+                    postalCode = Regex.Replace(postalCode, @"\d", ReplacementDigit);
+                }
+                else if (postalCode.Length >= InitialDigitsCount)
```

Comment:
❔ [Question]

Should we put the digitalCount and replacementDigit to settings as well?

Reply:
We can refine setting's fields later.